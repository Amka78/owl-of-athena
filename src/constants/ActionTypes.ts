const LOGIN = "LOGIN" as const;
const UPDATE_USER = "UPDATE_USER" as const;
const LOGOUT = "LOGOUT" as const;
const CACHE_SETTINGS = "CACHE_SETTINGS" as const;
const INIT = "INIT" as const;
const CACHE_PROFILES = "CACHE_PROFILES" as const;
const CACHE_SESSIONS = "CACHE_SESSIONS" as const;
const SELECT_SESSION = "SELECT_SESSION" as const;
const SELECT_SESSION_DETAIL = "SELECT_SESSION_DETAIL" as const;
const INITIALIZE_AURORA = "INITIALIZE_AURORA" as const;
const INITIALIZE_SESSION = "INITIALIZE_SESSION" as const;
const UPDATE_FILTER = "UPDATE_FILTER" as const;
const DELETE_SESSION = "DELETE_SESSION" as const;
const UPDATE_SESSION = "UPDATE_SESSION" as const;
export default {
    INIT,
    INITIALIZE_AURORA,
    INITIALIZE_SESSION,
    LOGIN,
    LOGOUT,
    UPDATE_USER,
    CACHE_SESSIONS,
    DELETE_SESSION,
    SELECT_SESSION,
    SELECT_SESSION_DETAIL,
    UPDATE_FILTER,
    UPDATE_SESSION,
    CACHE_SETTINGS,
    CACHE_PROFILES,
};
