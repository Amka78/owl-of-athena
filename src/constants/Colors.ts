const teal = "#00ffe6";
const purple = "#4a2fba";
const white = "#ffffff";
export default {
    black: "#000000",
    charcoal: "#333333",
    gray: "#9B9B9B",
    white: white,
    purple: purple,
    navy: "#070724",
    navy_darker: "#09092B",
    blue: "#001438",
    cyan: "#bcfeff",
    teal: teal,
    yellow: "#daf392",
    red: "#ff4310",
    light_orange: "#ff8260",
    semi_transparent: "#66000000",
    aurora_connected: teal,
    aurora_connected_text: purple,
    aurora_disconnected: purple,
    aurora_disconnected_text: white
};
