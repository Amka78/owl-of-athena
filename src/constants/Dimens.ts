export default {
    toolbar_title_text_size: 14,
    toolbar_button_text_size: 14,
    toolbar_button_margin_horizontal: 8,

    view_padding_bottom: 12,

    content_title_text_size: 24,
    content_title_margin_top: 32,
    content_title_margin_bottom: 48,
    content_text_size: 16,
    content_margin_horizontal: 48,

    snackbar_text_size: 8,

    statusbar_height: 8,

    navbar_height: 64,

    tip_text_size: 13,
    tip_text_margin_bottom: 13,

    error_text_size: 13,
    error_text_margin_bottom: 18,

    button_text_size: 24,
    button_height: 50,
    button_margin_bottom: 16,
    button_radius: 16,
    button_flat_text_size: 14,
    button_flat_margin_bottom: 12,
    button_flat_height: 28,

    input_text_size: 12,
    input_margin_bottom: 16,

    checkbox_text_size: 12,
    checkbox_outer_diameter: 24,
    checkbox_inner_diameter: 16,
    checkbox_gap: 4,
    checkbox_offset: 2,
    checkbox_padding_left: 16,
    checkbox_padding_right: 24,
    checkbox_margin_top: 16,
    checkbox_margin_bottom: 8,

    option_text_size: 14,
    option_height: 20,
    option_margin_bottom: 12,

    time_picker_height: 150,
    time_picker_scale: 1,
    time_picker_padding_horizontal: 0,

    circular_progress_diameter: 96,
    circular_progress_text_size: 32,

    // home fragment dimensions
    home_alarm_time_text_size: 48,
    home_alarm_meridian_text_size: 22,

    // session fragment dimensions
    session_label_text_size: 20,
    session_time_text_size: 28,
    session_sleep_score_text_size: 48,
    session_notes_text_size: 20,

    // session list fragment dimensions
    session_list_item_padding: 12,

    session_list_section_height: 24,
    session_list_section_padding_left: 24,
    session_list_section_text_size: 16,

    session_list_item_title_text_size: 20,
    session_list_item_subtitle_text_size: 15,
    session_list_item_text_margin_left: 18
};
