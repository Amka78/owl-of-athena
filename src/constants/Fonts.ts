export default {
    primaryRegular: "calibre_app_regular",
    primarySemiBold: "calibre_app_semibold"
};
