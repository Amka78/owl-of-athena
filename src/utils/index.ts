import BaseUrl from "./BaseUrl";
export { BaseUrl };
