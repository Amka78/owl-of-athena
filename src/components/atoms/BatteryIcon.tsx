//#region Import Modules
import React, { FunctionComponent } from "react";

import { Colors, Dimens } from "../../constants";
import { TemplateIcon, TemplateIconProps } from "./TemplateIcon";
//#endregion

//#region Types
export type BatteryIconProps = Omit<TemplateIconProps, "name" | "color"> & {
    batteryLevel: number;
    isUSBConnected: boolean;
};
//#endregion

//#region Component
export const BatteryIcon: FunctionComponent<BatteryIconProps> = (
    props: BatteryIconProps
) => {
    let batteryLevel = props.batteryLevel;
    if (props.batteryLevel === undefined || props.batteryLevel === null) {
        batteryLevel = 0;
    }
    let batteyIconSuffix = "";
    if (props.isUSBConnected) {
        batteyIconSuffix = "-charging";
    }

    let batterIcon = `battery${batteyIconSuffix}`;
    let color = Colors.aurora_connected;

    if (batteryLevel < 100) {
        if (batteryLevel <= 10) {
            color = Colors.red;
            batterIcon = "battery-alert";
        } else {
            batterIcon = `battery${batteyIconSuffix}-${batteryLevel
                .toString()
                .substring(0, 1)}0`;
        }
    }
    return (
        <TemplateIcon
            {...props}
            name={batterIcon}
            color={color}
            size={props.size ? props.size : Dimens.menu_icon_size}
        ></TemplateIcon>
    );
};
//#endregion
