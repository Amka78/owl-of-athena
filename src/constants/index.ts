import ActionTypes from "./ActionTypes";
import Colors from "./Colors";
import Dimens from "./Dimens";
import Fonts from "./Fonts";
import Layout from "./Layout";
import Message, { MessageLocalizationParam } from "./Message";
import { MessageKeys } from "./MessageKeys";
import StorageKeys from "./StorageKeys";

export {
    ActionTypes,
    Colors,
    Dimens,
    Fonts,
    Layout,
    MessageLocalizationParam,
    Message,
    MessageKeys,
    StorageKeys
};
