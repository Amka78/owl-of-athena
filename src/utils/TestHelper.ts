import "../components";

import { shallow, ShallowWrapper } from "enzyme";
import enzymetoJson from "enzyme-to-json";
import {
    GestureResponderEvent,
    NativeSyntheticEvent,
    TextInputEndEditingEventData,
} from "react-native";

import { ButtonProps } from "../components/atoms/Button";

// @ts-ignore
export const FakeGestureResponderEvent: GestureResponderEvent = {};
// @ts-ignore
export const FakeTextBoxOnEndEditingEventData: NativeSyntheticEvent<TextInputEndEditingEventData> = {};

export function createMock<P, S>(comp: any): ShallowWrapper<P, S> {
    return shallow<P, S>(comp);
}

export function toJson(wrapper: any) {
    return enzymetoJson(wrapper);
}

/**
 * TestIDを利用してComponetのPropsを取得します。
 *
 * @static
 * @template T Propsの型を指定
 * @param {ShallowWrapper} component
 * @param {string} name
 * @param {string} testId
 * @returns {T}
 * @memberof TestHelper
 */
export function getComponentPropsByTestID<T>(
    component: ShallowWrapper<any, any>,
    name: string,
    testId: string
): T | undefined {
    const findComponent = component.find(name).findWhere((wrapper: any) => {
        const targetProps = wrapper.props();
        return targetProps.testID === testId;
    });
    if (findComponent.length <= 0) {
        return undefined;
    }
    const props: T = findComponent.props();
    return props;
}

export function getButtonPropsByTestID(
    component: ShallowWrapper<any, any>,
    testID: string
): ButtonProps | undefined {
    return getComponentPropsByTestID<ButtonProps>(
        component,
        "MultiLingualButton",
        testID
    );
}

/*public static getTextBoxPropsByTestID(
        component: ShallowWrapper<any, any>,
        testID: string
    ): ITextBoxProps | undefined {
        return this.getComponentPropsByTestID<ITextBoxProps>(
            component,
            "MultiLingualTextBox",
            testID
        );
    }

    public static getHeaderSearchBarPropsByTestID(
        component: ShallowWrapper<any, any>,
        testID: string
    ): IHeaderSearchBarProps | undefined {
        return this.getComponentPropsByTestID<IHeaderSearchBarProps>(
            component,
            "HeaderSearchBar",
            testID
        );
    }

    public static getListItemPropsByTestID(
        component: ShallowWrapper<any, any>,
        testID: string
    ): IListItemProps | undefined {
        return this.getComponentPropsByTestID<IListItemProps>(
            component,
            "MultiLingualListItem",
            testID
        );
    }*/

/*public static createNavigationMock(
        param: Partial<NavigationScreenProp<any, any>>
    ): NavigationScreenProp<any, any> {
        const navigation = new Object();
        Object.assign(navigation, param);
        return navigation as NavigationScreenProp<any, any>;
    }*/
