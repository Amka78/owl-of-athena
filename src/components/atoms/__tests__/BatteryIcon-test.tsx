//#region Import Modules
import { ShallowWrapper } from "enzyme";
import React from "react";

import { Colors } from "../../../constants";
import { createMock, toJson } from "../../../utils/TestHelper";
import { BatteryIcon, BatteryIconProps } from "../BatteryIcon";
//#endregion

//#region Types
type TestPairs = {
    batteryLevel?: number | null;
    isUSBConnected: boolean;

    resultIcon:
        | "battery-alert"
        | "battery"
        | "battery-charging"
        | "battery-10"
        | "battery-charging-10"
        | "battery-20"
        | "battery-charging-20"
        | "battery-30"
        | "battery-charging-30"
        | "battery-40"
        | "battery-charging-40"
        | "battery-50"
        | "battery-charging-50"
        | "battery-60"
        | "battery-charging-60"
        | "battery-70"
        | "battery-charging-70"
        | "battery-80"
        | "battery-charging-80"
        | "battery-90"
        | "battery-charging-90";
    resultIconColor: string;
};
//#endregion
//#region Tests
let component: ShallowWrapper<BatteryIconProps, unknown, unknown>;
describe("BatteryIcon UnitTest", () => {
    it("Renders correctly", () => {
        component = createMock(
            <BatteryIcon batteryLevel={100} isUSBConnected={true}></BatteryIcon>
        );

        expect(toJson(component)).toMatchSnapshot();
    });

    it.each<TestPairs>([
        {
            batteryLevel: undefined,
            isUSBConnected: false,
            resultIcon: "battery-alert",
            resultIconColor: Colors.red,
        },
        {
            batteryLevel: null,
            isUSBConnected: false,
            resultIcon: "battery-alert",
            resultIconColor: Colors.red,
        },
        {
            batteryLevel: undefined,
            isUSBConnected: true,
            resultIcon: "battery-alert",
            resultIconColor: Colors.red,
        },
        {
            batteryLevel: null,
            isUSBConnected: true,
            resultIcon: "battery-alert",
            resultIconColor: Colors.red,
        },
        {
            batteryLevel: 0,
            isUSBConnected: false,
            resultIcon: "battery-alert",
            resultIconColor: Colors.red,
        },
        {
            batteryLevel: 0,
            isUSBConnected: false,
            resultIcon: "battery-alert",
            resultIconColor: Colors.red,
        },
        {
            batteryLevel: 9,
            isUSBConnected: false,
            resultIcon: "battery-alert",
            resultIconColor: Colors.red,
        },
        {
            batteryLevel: 9,
            isUSBConnected: false,
            resultIcon: "battery-alert",
            resultIconColor: Colors.red,
        },
        {
            batteryLevel: 10,
            isUSBConnected: false,
            resultIcon: "battery-10",
            resultIconColor: Colors.aurora_connected,
        },
        {
            batteryLevel: 10,
            isUSBConnected: true,
            resultIcon: "battery-charging-10",
            resultIconColor: Colors.aurora_connected,
        },
        {
            batteryLevel: 19,
            isUSBConnected: false,
            resultIcon: "battery-10",
            resultIconColor: Colors.aurora_connected,
        },
        {
            batteryLevel: 19,
            isUSBConnected: true,
            resultIcon: "battery-charging-10",
            resultIconColor: Colors.aurora_connected,
        },
        {
            batteryLevel: 20,
            isUSBConnected: false,
            resultIcon: "battery-20",
            resultIconColor: Colors.aurora_connected,
        },
        {
            batteryLevel: 20,
            isUSBConnected: true,
            resultIcon: "battery-charging-20",
            resultIconColor: Colors.aurora_connected,
        },
        {
            batteryLevel: 29,
            isUSBConnected: false,
            resultIcon: "battery-20",
            resultIconColor: Colors.aurora_connected,
        },
        {
            batteryLevel: 29,
            isUSBConnected: true,
            resultIcon: "battery-charging-20",
            resultIconColor: Colors.aurora_connected,
        },
        {
            batteryLevel: 30,
            isUSBConnected: false,
            resultIcon: "battery-30",
            resultIconColor: Colors.aurora_connected,
        },
        {
            batteryLevel: 30,
            isUSBConnected: true,
            resultIcon: "battery-charging-30",
            resultIconColor: Colors.aurora_connected,
        },
        {
            batteryLevel: 39,
            isUSBConnected: false,
            resultIcon: "battery-30",
            resultIconColor: Colors.aurora_connected,
        },
        {
            batteryLevel: 39,
            isUSBConnected: true,
            resultIcon: "battery-charging-30",
            resultIconColor: Colors.aurora_connected,
        },
        {
            batteryLevel: 40,
            isUSBConnected: false,
            resultIcon: "battery-40",
            resultIconColor: Colors.aurora_connected,
        },
        {
            batteryLevel: 40,
            isUSBConnected: true,
            resultIcon: "battery-charging-40",
            resultIconColor: Colors.aurora_connected,
        },
        {
            batteryLevel: 49,
            isUSBConnected: false,
            resultIcon: "battery-40",
            resultIconColor: Colors.aurora_connected,
        },
        {
            batteryLevel: 49,
            isUSBConnected: true,
            resultIcon: "battery-charging-40",
            resultIconColor: Colors.aurora_connected,
        },
        {
            batteryLevel: 50,
            isUSBConnected: false,
            resultIcon: "battery-50",
            resultIconColor: Colors.aurora_connected,
        },
        {
            batteryLevel: 50,
            isUSBConnected: true,
            resultIcon: "battery-charging-50",
            resultIconColor: Colors.aurora_connected,
        },
        {
            batteryLevel: 59,
            isUSBConnected: false,
            resultIcon: "battery-50",
            resultIconColor: Colors.aurora_connected,
        },
        {
            batteryLevel: 59,
            isUSBConnected: true,
            resultIcon: "battery-charging-50",
            resultIconColor: Colors.aurora_connected,
        },
        {
            batteryLevel: 60,
            isUSBConnected: false,
            resultIcon: "battery-60",
            resultIconColor: Colors.aurora_connected,
        },
        {
            batteryLevel: 60,
            isUSBConnected: true,
            resultIcon: "battery-charging-60",
            resultIconColor: Colors.aurora_connected,
        },
        {
            batteryLevel: 69,
            isUSBConnected: false,
            resultIcon: "battery-60",
            resultIconColor: Colors.aurora_connected,
        },
        {
            batteryLevel: 69,
            isUSBConnected: true,
            resultIcon: "battery-charging-60",
            resultIconColor: Colors.aurora_connected,
        },
        {
            batteryLevel: 70,
            isUSBConnected: false,
            resultIcon: "battery-70",
            resultIconColor: Colors.aurora_connected,
        },
        {
            batteryLevel: 70,
            isUSBConnected: true,
            resultIcon: "battery-charging-70",
            resultIconColor: Colors.aurora_connected,
        },
        {
            batteryLevel: 79,
            isUSBConnected: false,
            resultIcon: "battery-70",
            resultIconColor: Colors.aurora_connected,
        },
        {
            batteryLevel: 79,
            isUSBConnected: true,
            resultIcon: "battery-charging-70",
            resultIconColor: Colors.aurora_connected,
        },
        {
            batteryLevel: 80,
            isUSBConnected: false,
            resultIcon: "battery-80",
            resultIconColor: Colors.aurora_connected,
        },
        {
            batteryLevel: 80,
            isUSBConnected: true,
            resultIcon: "battery-charging-80",
            resultIconColor: Colors.aurora_connected,
        },
        {
            batteryLevel: 89,
            isUSBConnected: false,
            resultIcon: "battery-80",
            resultIconColor: Colors.aurora_connected,
        },
        {
            batteryLevel: 89,
            isUSBConnected: true,
            resultIcon: "battery-charging-80",
            resultIconColor: Colors.aurora_connected,
        },
        {
            batteryLevel: 90,
            isUSBConnected: false,
            resultIcon: "battery-90",
            resultIconColor: Colors.aurora_connected,
        },
        {
            batteryLevel: 90,
            isUSBConnected: true,
            resultIcon: "battery-charging-90",
            resultIconColor: Colors.aurora_connected,
        },
        {
            batteryLevel: 99,
            isUSBConnected: false,
            resultIcon: "battery-90",
            resultIconColor: Colors.aurora_connected,
        },
        {
            batteryLevel: 99,
            isUSBConnected: true,
            resultIcon: "battery-charging-90",
            resultIconColor: Colors.aurora_connected,
        },
        {
            batteryLevel: 100,
            isUSBConnected: false,
            resultIcon: "battery",
            resultIconColor: Colors.aurora_connected,
        },
        {
            batteryLevel: 100,
            isUSBConnected: true,
            resultIcon: "battery-charging",
            resultIconColor: Colors.aurora_connected,
        },
    ])(
        "The icons and colours must correspond to the battery display",
        (props: TestPairs) => {
            component = createMock(
                <BatteryIcon
                    batteryLevel={props.batteryLevel as any}
                    isUSBConnected={props.isUSBConnected}
                ></BatteryIcon>
            );
            expect(component.props().name).toBe(props.resultIcon);
            expect(component.props().color).toBe(props.resultIconColor);
        }
    );

    it("Overwrite default props value.", () => {
        component = createMock(
            <BatteryIcon
                batteryLevel={100}
                isUSBConnected={true}
                size={50}
            ></BatteryIcon>
        );
        expect(component.props().size).toBe(50);
    });
});
//#endregion
