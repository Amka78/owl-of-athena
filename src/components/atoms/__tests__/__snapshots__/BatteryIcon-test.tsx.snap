// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BatteryIcon UnitTest renders correctly 1`] = `
<Component
  batteryLevel={100}
  color="#00ffe6"
  isUSBConnected={true}
  name="battery-charging"
  size={40}
/>
`;
